### ISSUE TITLE
Navigation Panel or HUD

### ISSUE THREAD

**Issue by diethardsteiner** [uranusjr/macdown#312]
I work with very long documents. Currently it is very difficult to find the right section within the document. It would be great if you could add a navigation panel or HUD which would allow me to jump to a specific section within the document. The navigation panel should display all the headers of the document hierarchically and should allow me to collapse certain levels if needed.

**Comment by ohthetrees** [comment]
A workaround is to put `[toc]` at the top of your document, then in the menu select View-->Renders TOC

That will render a table of contents, though it is rendered in the preview pane, rather than a navigation panel, and it does not support collapsing the hierarchy.

Not what you were asking for, but it might get you half way there.

**Comment by diethardsteiner** [comment]
Thanks for pointing this out - I was aware of this feature. Unfortunately it doesn't allow me to jump to the right section in the editor.

**Comment by ohthetrees** [comment]
You are right, the editor does not stay synchronized when you click a link in the preview window. That would certainly be nice.

**Comment by uranusjr** [comment]
Jumping in editor is currently impossible (at least without very dirty hacks) due to the internal rendering mechanism.

**Comment by diethardsteiner** [comment]
Does this mean it will never be implemented? I've been hoping for this feature to be released for quite some time. Actually this is the only feature missing to make MacDown my only markdown editor.

**Comment by uranusjr** [comment]
Actually the rendering engine is currently being rewritten. There are still some way to go before it is done, and even after that it will take some time—something I severely lack recently—to integrate the new structure into MacDown, and then to build this specific feature. It can be done, just don’t hold your breath.

**Comment by diethardsteiner** [comment]
Ok, thanks for letting me know!
